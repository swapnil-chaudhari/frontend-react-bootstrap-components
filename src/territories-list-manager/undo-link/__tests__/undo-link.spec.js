import React from 'react';
import renderShallow from 'render-shallow';
import { findWithClass } from 'react-shallow-testutils';
import UndoLink from '../undo-link';

describe('<UndoLink>', () => {
    describe('when it renders', () => {
        let component;
        const props = {
            handleUndoChanges: jest.fn(),
            handleNotificationClose: jest.fn(),
            movedCountriesNumber: 1,
            formatMessage: msg => msg,
            messages: {
                undoDescription: 'undo description',
                undoLabel: 'undo label'
            }
        };

        beforeAll(() => {
            component = renderShallow(<UndoLink { ...props } />).output;
        });

        test('renders correct class name', () => {
            expect(component.props.className).toEqual('UndoLink');
        });

        test('renders correct number of countries', () => {
            expect(component.props.children[0]).toMatchSnapshot();
        });

        test('should call handleUndoChanges on UNDO link click', () => {
            const undoLink = findWithClass(component, 'UndoLink-link');

            undoLink.props.onClick();

            expect(props.handleUndoChanges).toHaveBeenCalled();
        });

        test('should call handleNotificationClose on close icon click', () => {
            const closeIcon = component.props.children[1];
            closeIcon.props.onClick();

            expect(props.handleNotificationClose).toHaveBeenCalled();
        });
    });
});
