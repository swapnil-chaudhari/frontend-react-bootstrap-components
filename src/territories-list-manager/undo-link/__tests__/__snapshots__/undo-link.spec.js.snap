// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UndoLink> when it renders renders correct number of countries 1`] = `
<span>
  <span>
    1 undo description
  </span>
  <span
    className="UndoLink-link btn btn-link"
    onClick={[MockFunction]}
  >
    undo label
  </span>
</span>
`;
